django/contrib/gis/geos/geometry.py

```diff
@@ -163,6 +164,27 @@ def __setstate__(self, state):
     def _from_wkb(cls, wkb):
         return wkb_r().read(wkb)
 
+    @staticmethod
+    def from_ewkt(ewkt):
+        if not ewkt:
+            return ValueError('EWKT is expected, but empty string is given.')
+        ewkt = force_bytes(ewkt)
+        srid = None
+        parts = ewkt.split(b';', 1)
+        if len(parts) == 2:
+            srid_part, wkt = parts
+            match = re.match(b'SRID=(?P<srid>\-?\d+)', srid_part)
+            if not match:
+                raise ValueError('EWKT has invalid SRID part.')
```

Comment:
Worth some unit tests for `from_ewkt()`? This line isn't tested based on the coverage report for PostGIS and if the other exception raising in this method is removed, the only failure is in `gis_tests.test_geoforms.GeometryFieldTest.test_field_string_value` which wouldn't be very easy to track down, I'd think.